src/Dialect/ONNX/ShapeInference/Gather.cpp

```diff
@@ -14,59 +18,21 @@ using namespace mlir;
 
 namespace onnx_mlir {
 
-ONNXGatherOpShapeHelper::ONNXGatherOpShapeHelper(
-    ONNXGatherOp *newOp, IndexExprScope *inScope)
-    : ONNXOpShapeHelper<ONNXGatherOp>(
-          newOp, newOp->getOperation()->getNumResults(), inScope),
-      dataDims(), indicesDims(), positiveConstantIndices(false) {}
-
-ONNXGatherOpShapeHelper::ONNXGatherOpShapeHelper(ONNXGatherOp *newOp,
-    OpBuilder *rewriter, ArrayValueIndexCapture::GetDenseVal fGetDenseVal,
-    ArrayValueIndexCapture::LoadVal fLoadVal, IndexExprScope *inScope)
-    : ONNXOpShapeHelper<ONNXGatherOp>(newOp,
-          newOp->getOperation()->getNumResults(), rewriter, fGetDenseVal,
-          fLoadVal, inScope),
-      dataDims(), indicesDims(), positiveConstantIndices(false) {}
-
 LogicalResult ONNXGatherOpShapeHelper::computeShape(
     ONNXGatherOpAdaptor operandAdaptor) {
   // Shape inference indicated by passing a null rewriter pointer.
   // Read data and indices shapes as dim indices.
   MemRefBoundsIndexCapture dataBounds(operandAdaptor.data());
   MemRefBoundsIndexCapture indicesBounds(operandAdaptor.indices());
+  DimsExpr dataDims, indicesDims;
   dataBounds.getDimList(dataDims);
   indicesBounds.getDimList(indicesDims);
 
-  // Read constant 'axis' attribute and normalize when negative.
+  int64_t dataRank = dataDims.size();
   int64_t axisIndex = op->axis();
-  // The 'axis' value must be in [-rank, rank-1].
```

Comment:
Simplified to make more consistent with other shape helpers.  The auxiliary vector (dataDims) now computed in the ONNXToKrnl conversion code.